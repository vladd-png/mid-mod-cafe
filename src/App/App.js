import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form.js';
import ReservationContainer from '../ReservationContainer/ReservationContainer.js';
import { getReservations, addReservation, removeReservation } from '../apiCalls.js'


class App extends Component {
  constructor() {
    super()
    this.state = {
      reservations: []
    }
  }
  componentDidMount = () => {
    getReservations()
      .then(reservations =>  this.setState({ reservations }))
      .catch(error => console.log(error.message))
  }
  addReservation = reservation => {
    addReservation(reservation)
      .then(data => this.setState({ reservations: [...this.state.reservations, data] }))
  }
  cancelReservation = id => {
    let filteredReso = this.state.reservations.filter(reso => reso.id !== id)
    this.setState({ reservations: filteredReso })
    removeReservation(id)
      .then(response => console.log(response))
  }
  sortReservation = props => {
    if (props === 'first') {
      let sorted = this.state.reservations.sort((a, b) => parseInt(b.date) - parseInt(a.date))
    } else if (props === 'last') {
      let sorted = this.state.reservations.sort((a, b) => parseInt(a.date) - parseInt(b.date))
    }
  }
  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form addReservation={this.addReservation}/>
        </div>
        <div className='resy-container'>
          <ReservationContainer reservations={this.state.reservations} cancelReservation={this.cancelReservation} sortReservation={this.sortReservation}/>
        </div>
      </div>
    )
  }
}

export default App;
