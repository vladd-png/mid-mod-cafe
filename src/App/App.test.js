import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper=shallow(<App />);
  })

  it('should be an instance of App', () => {
    expect(wrapper).toMatchSnapshot();
  })

})
