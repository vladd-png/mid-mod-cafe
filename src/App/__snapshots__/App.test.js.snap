// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be an instance of App 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationContainer
      reservations={Array []}
    />
  </div>
</div>
`;
